test/cpp/jit/backend_with_compiler.cpp

```diff
@@ -0,0 +1,111 @@
+#include <torch/csrc/jit/backends/backend.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+std::vector<std::string> parseBlob(const std::string& blob) {
+  std::vector<std::string> result;
+  std::stringstream s_stream(blob);
+  while (s_stream.good()) {
+    std::string substr;
+    getline(s_stream, substr, ',');
+    result.push_back(substr);
+  }
+  return result;
+}
+} // namespace
+// This test JIT backend is intended to do the minimal amount of work
+// necessary to test that the JIT backend registration endpoints and
+// code generation are working correctly. It is not intended to
+// produce numerically correct results.
+class BackendWithCompiler : public PyTorchBackendInterface {
+ public:
+  // Constructor.
+  explicit BackendWithCompiler() {}
+  virtual ~BackendWithCompiler() = default;
+
+  c10::impl::GenericDict compile(
+      c10::IValue processed,
+      c10::impl::GenericDict method_compile_spec) override {
+    return processed.toGenericDict();
+  }
+
+  c10::impl::GenericList execute(
+      c10::IValue handle,
+      c10::impl::GenericList inputs) override {
+    TORCH_INTERNAL_ASSERT(handle.isString());
+    TORCH_INTERNAL_ASSERT(inputs.size() == 2);
+    c10::IValue val0 = inputs[0];
+    at::Tensor x = val0.toTensor();
+    c10::IValue val1 = inputs[1];
+    at::Tensor h = val1.toTensor();
+
+    c10::List<at::Tensor> output_list;
+    auto tokens = parseBlob(handle.toStringRef());
+    for (const auto& token : tokens) {
+      if (token == "aten::add") {
+        output_list.emplace_back(x.add_(h, 1.0));
+      } else if (token == "aten::sub") {
+        output_list.emplace_back(x.sub_(h, 1.0));
+      }
+    }
+    return c10::impl::toList(output_list);
+  }
+};
+
+// Actual AOT compilation function. Put here for demonstration of backend
+// as a whole piece. It's used when compilation is required. A dummy function
+// can be passed when there's no usage of compilation in runtime backend lib.
```

Comment:
I'd just clarify: "there's no usage of AOT compilation " since "compilation" could still theoretically take place on-target.

I'll clarify/add the docs to these interfaces.